src/batch.jl

```diff
@@ -32,14 +57,17 @@ end
 #   @cfunction($bc, Cvoid, (Ptr{UInt},))
 # end
 
+
 @inline function setup_batch!(
   p::Ptr{UInt},
   fptr::Ptr{Cvoid},
+  closure_obj,
```

Comment:
Why would we want both `fptr` and `closure_obj`? Wouldn't it be one or the other?

Reply:
`fptr` allows to infer the type of `BatchClosure{F,A,C}` thanks to the `@cfunction`.
We can't place the closure object directly into the TLS because it contains the captured values in a `Core.Box`, which doesn't have a fixed size at compile-time, and therefore `store!` fails.
So I put the closure in a `Reference`, which does have a fixed size, and when calling `FakeClosure{F,A,C}` we have the type of the closure *before* `load`-ing it, allowing type-inference.